src/resources/js/store/localisation.js

```diff
@@ -6,11 +6,32 @@ export const state = {
 
 export const getters = {
     isInitialised: state => Object.keys(state.i18n).length > 0,
-    __: (state, getters, rootState, rootGetters) => (key) => {
-        const lang = rootGetters['preferences/lang'];
-        return state.i18n[lang]
+    __: (state, getters, rootState, rootGetters) => (key, params) => {
+        const lang = rootGetters['preferences/lang']
+            || ((rootState.preferences && rootState.preferences.global)
+                ? rootState.preferences.global.lang : null);
+
+        let translation = state.i18n[lang]
             ? state.i18n[lang][key]
             : key;
+
+        if (typeof translation === 'undefined'
+            && rootState.localisation.keyCollector) {
+                store.dispatch('localisation/addMissingKey', key);
+        }
```

Comment:
why should we call a mutation / action from a getter ?

Reply:
As the _global_ helper isn't available within the vue store context itself (we have a few translatable strings directly in there as they're being used for _global_ dialogs)

Reply:
I wasn't clear enough. We won't call mutations nor actions from getters. We will use the global helpers as it is, and if it is needed outside a vue instance we can import it.

Reply:
I understand, however see below for the implications